core/src/main/java/com/linecorp/armeria/server/DecoratingServiceBindingBuilder.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.MediaType;
+
+/**
+ * A builder class for binding a {@code decorator} with {@link Route} fluently.
+ * This class can be instantiated through {@link ServerBuilder#routeDecorator()}.
+ *
+ * <p>Call {@link #build(Function)} or {@link #build(DecoratingServiceFunction)}
+ * to build the {@code decorator} and return to the {@link ServerBuilder}.
+ *
+ * <pre>{@code
+ * ServerBuilder sb = new ServerBuilder();
+ *
+ * sb.routeDecorator()                                // Configure a decorator with route.
+ *   .pathUnder("/api/users")
+ *   .build((delegate, ctx, req) -> {
+ *       if (!AUTHORIZATION.equals(req.headers().get("Authorization"))) {
+ *           return HttpResponse.of(HttpStatus.UNAUTHORIZED);
+ *       }
+ *       return delegate.serve(ctx, req);
+ *   });                                              // Return to the ServerBuilder.
+ * }</pre>
+ *
+ * @see ServiceBindingBuilder
```

Comment:
Did you mean `@see VirtualHostDecoratingServiceBindingBuilder`?

Reply:
Thanks!